src/include/daos_prop.h

```diff
@@ -347,7 +347,17 @@ struct daos_prop_entry {
 /** Allowed max number of property entries in daos_prop_t */
 #define DAOS_PROP_ENTRIES_MAX_NR	(128)
 /** max length for pool/container label */
-#define DAOS_PROP_LABEL_MAX_LEN		(256)
+
+/* DAOS labels (pool/container properties) must start with an
+ * a-zA-Z, then consist only of alphanumeric characters, colon ':',
+ * period '.' or underscore '_', and must be of length
+ * 1 - DAOS_PROP_LABEL_MAX_LEN.
+ */
+#define DAOS_LABEL_REGEX "([a-zA-Z])+([a-zA-Z0-9._:]+)"
```

Comment:
discussion in working group that we don't need to constrain the first character to be alphabetical. Will try to align this exactly with the WIP libduns patch https://github.com/daos-stack/daos/pull/5962 that currently defines a regex named LABEL_REGEX.

After this PR and PR-5962 land, then we can prepare another patch to align on the single DAOS_LABEL_REGEX definition.